### PR TITLE
Add scanner action that adds restrictions on top of delaying auto-approval

### PR DESCRIPTION
**Pull request by diox**
Contains a couple changes that go with that:
- Keep stale add-ons and fileuploads for a little while longer
- Prevent using `FileUpload` belonging to another add-on to create a `Version`
- Link the `FileUpload` to the `Addon` after creating a Version (if none was set)

Fixes https://github.com/mozilla/addons-server/issues/16504

### CODE REVIEW

File: src/olympia/scanners/actions.py [hunk 1]
```diff
@@ -37,10 +38,37 @@ def _delay_auto_approval_indefinitely(version):
...
+
+    for ip in ips:
+        IPNetworkUserRestriction.objects.get_or_create(network=f'{ip}/32')
+
+
 def _flag_for_human_review_by_scanner(version, scanner):
     from olympia.versions.models import VersionReviewerFlags
 
-    if scanner is not MAD:
+    if scanner != MAD:
```
Comment (diox) [inline src/olympia/scanners/actions.py:71]: Small drive-by, this was wrong and just worked by accident, identity != equality

**Review by willdurand — APPROVED** [review]
